packages/mdc-dialog/index.ts

```diff
@@ -21,167 +21,11 @@
-      document.addEventListener('keydown', this.handleDocumentKeydown_);
-    };
-    this.handleClosing_ = () => {
-      LAYOUT_EVENTS.forEach((evtType) => window.removeEventListener(evtType, this.handleLayout_));
-      document.removeEventListener('keydown', this.handleDocumentKeydown_);
-    };
-
-    this.listen('click', this.handleInteraction_);
-    this.listen('keydown', this.handleInteraction_);
-    this.listen(strings.OPENING_EVENT, this.handleOpening_);
-    this.listen(strings.CLOSING_EVENT, this.handleClosing_);
-  }
-
-  destroy() {
-    this.unlisten('click', this.handleInteraction_);
-    this.unlisten('keydown', this.handleInteraction_);
-    this.unlisten(strings.OPENING_EVENT, this.handleOpening_);
-    this.unlisten(strings.CLOSING_EVENT, this.handleClosing_);
-    this.handleClosing_();
-
-    this.buttonRipples_.forEach((ripple) => ripple.destroy());
-    super.destroy();
-  }
-
-  layout() {
-    this.foundation_.layout();
-  }
-
-  open() {
-    this.foundation_.open();
-  }
-
-  close(action = '') {
-    this.foundation_.close(action);
-  }
-
-  getDefaultFoundation() {
-    return new MDCDialogFoundation({
-      addBodyClass: (className) => document.body.classList.add(className),
-      addClass: (className) => this.root_.classList.add(className),
-      areButtonsStacked: () => util.areTopsMisaligned(this.buttons_),
-      clickDefaultButton: () => this.defaultButton_ && this.defaultButton_.click(),
-      eventTargetMatches: (target, selector) => target ? matches(target as Element, selector) : false,
-      getActionFromEvent: (evt: Event) => {
-        if (!evt.target) {
-          return '';
-        }
-        const element = closest(evt.target as Element, `[${strings.ACTION_ATTRIBUTE}]`);
-        return element && element.getAttribute(strings.ACTION_ATTRIBUTE);
-      },
-      hasClass: (className) => this.root_.classList.contains(className),
-      isContentScrollable: () => util.isScrollable(this.content_),
-      notifyClosed: (action) => this.emit(strings.CLOSED_EVENT, action ? {action} : {}),
-      notifyClosing: (action) => this.emit(strings.CLOSING_EVENT, action ? {action} : {}),
-      notifyOpened: () => this.emit(strings.OPENED_EVENT, {}),
-      notifyOpening: () => this.emit(strings.OPENING_EVENT, {}),
-      releaseFocus: () => this.focusTrap_.deactivate(),
-      removeBodyClass: (className) => document.body.classList.remove(className),
-      removeClass: (className) => this.root_.classList.remove(className),
-      reverseButtons: () => {
-        this.buttons_.reverse();
-        this.buttons_.forEach((button) => {
-          button.parentElement!.appendChild(button);
-        });
-      },
-      trapFocus: () => this.focusTrap_.activate(),
-    });
-  }
-}
-
-export {MDCDialog as default, MDCDialog, util};
+export {MDCDialog as default} from './component';
 export * from './adapter';
+export * from './component';
 export * from './foundation';
 export * from './types';
+export {util};
```

Comment:
Should this be moved up? Seems weird to have `export *`s sandwiched between two more specific exports. Or is the idea here that the non-`from` export is last?